crates/iota-indexer/tests/common/mod.rs

```diff
@@ -114,7 +114,7 @@ pub async fn start_test_cluster_with_read_write_indexer(
     builder_modifier: Option<Box<dyn FnOnce(TestClusterBuilder) -> TestClusterBuilder>>,
 ) -> (TestCluster, PgIndexerStore<PgConnection>, HttpClient) {
     let temp = tempdir().unwrap().into_path();
-    let mut builder = TestClusterBuilder::new().with_data_ingestion_dir(temp.clone());
+    let mut builder = TestClusterBuilder::new();
```

Comment:
This is not necessary to force the indexer syncing from remote, so why remove it?

Reply:
I'm afraid in order to force the sync only trough the REST API we need to remove it. I did a quick test with and without the data ingestion path and rarely it fetches form REST API since all the checkpoints are available from the validators and the fullnode is always behind.

these are the steps it does:

1.  The [synchronization](https://github.com/iotaledger/iota/blob/develop/crates/iota-data-ingestion-core/src/reader.rs#L369-L371) process is triggered either when an inotify notification is received from monitoring the data ingestion path or according to the specified tick interval (`self.options.tick_interval_ms`).

2. Within the [synchronization](https://github.com/iotaledger/iota/blob/develop/crates/iota-data-ingestion-core/src/reader.rs#L245-L251) routine, the process first checks for any available checkpoints in the data ingestion path. If a [remote](https://github.com/iotaledger/iota/blob/develop/crates/iota-data-ingestion-core/src/reader.rs#L254-L260) source is specified, the system also fetches checkpoints from that source, allowing for the merging of checkpoints obtained from both origins.

3. The issue is the fact that checkpoints in the data ingestion path are generated more rapidly than those retrieved from the full node. Validators can produce checkpoints at a faster rate, while the full node must synchronize with these validators to obtain the latest checkpoints. Consequently, when the system performs remote fetching, it may not yield the most current checkpoints. Thus, checkpoints written into the data ingestion path consistently remain ahead of those accessible via the full node’s REST API.

4. By removing the data ingestion path from the `TestClusterBuilder`, nodes will stop writing checkpoints to that directory. As a result, the only mechanism for synchronization available to the indexer will be the remote fetch. This situation will lead to an empty vector for checkpoints from the data ingestion path

Reply:
You're right, I missed the `&&` condition.